tests/kernel/mem_protect/userspace/prj.conf

```diff
@@ -1,3 +1,4 @@
 CONFIG_ZTEST=y
 CONFIG_USERSPACE=y
 CONFIG_APPLICATION_MEMORY=y
+CONFIG_ASSERT=y
```

Comment:
Do we want this enabled for these tests?

Reply:
CONFIG_ASSERT is enabled for the test thread_in_multiple_domains(), which  checks for assert condition in k_mem_domain_add() API. I agree that the test should belong to set of tests related to memory domains in #5197. I will move this test once #5197 gets merged.

Reply:
CONFIG_ASSERT is enabled by default with sanitycheck

Reply:
@nashif But to execute the thread_in_multiple_domains() locally, we may need this. The test is meant to hit assert failure condition. If we do not enable it, the test case will fail without checking for assertion when run locally.

Reply:
The problem with CONFIG_ASSERT is that when an assertion hits, it just hangs the system instead of having ztest report success or failure.
Is there any other way we can detect this situation in the test itself and fail it accordingly?

Reply:
I agree. But here in this test, ztest thread is creating another thread and this thread hits assertion. The ztest thread then checks if the assert condition is hit and reports the test result.